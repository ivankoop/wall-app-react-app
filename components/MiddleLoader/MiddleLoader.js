import React, {Component} from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {
    View,
    StyleSheet,
    ActivityIndicator,
    Text,
    Dimensions,
} from "react-native";

const {width} = Dimensions.get("window");

class MiddleLoader extends Component {
    constructor(props) {
        super(props);
    }

    static propTypes = {
        visibleLoader: PropTypes.bool,
    };

    render() {
        const {visibleLoader} = this.props;

        return (
            visibleLoader && (
                <View style={styles.mainContainer}>
                    <View style={styles.contentContainer}>
                        <ActivityIndicator
                            style={styles.indicator}
                            size="large"
                            color="#58F1FF"
                        />
                        <Text style={styles.text}>Loading...</Text>
                    </View>
                </View>
            )
        );
    }
}

const mapStateToProps = state => {
    const {visibleLoader} = state.middleLoaderControls;

    return {
        visibleLoader: visibleLoader,
    };
};

export default connect(mapStateToProps)(MiddleLoader);

const styles = StyleSheet.create({
    mainContainer: {
        position: "absolute",
        width: "100%",
        height: "100%",
        backgroundColor: "transparent",
        justifyContent: "center",
    },

    contentContainer: {
        alignSelf: "center",
        position: "absolute",
        backgroundColor: "rgba(0,0,0,0.8)",
        width: width * 0.4,
        paddingTop: 20,
        paddingBottom: 20,
        borderRadius: 63,
        paddingRight: 10,
        paddingLeft: 10,
        flexDirection: "row",
    },

    indicator: {
        flex: 0.2,
        marginLeft: 15,
    },

    text: {
        color: "white",
        flex: 1,
        alignSelf: "center",
        paddingLeft: 20,
        fontSize: width * 0.04,
        fontFamily: "Roboto_medium",
    },
});
