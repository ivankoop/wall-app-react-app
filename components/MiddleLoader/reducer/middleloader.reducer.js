import {
    SHOW_MIDDLE_LOADER,
    HIDE_MIDDLE_LOADER,
} from "../constants/middleloader.constants";

const INTIAL_STATE = {
    visibleLoader: false,
};

export function middleLoaderControls(state = INTIAL_STATE, action) {
    switch (action.type) {
        case SHOW_MIDDLE_LOADER:
            return {
                ...state,
                visibleLoader: true,
            };
        case HIDE_MIDDLE_LOADER:
            return {
                ...state,
                visibleLoader: false,
            };

        default:
            return state;
    }
}
