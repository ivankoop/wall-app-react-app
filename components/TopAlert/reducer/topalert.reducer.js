import {SHOW_TOP_ALERT, HIDE_TOP_ALERT} from "../constants/topalert.constants";

const INTIAL_STATE = {
    alertType: "warning",
    text: "",
    visible: false,
};

export function topAlertControls(state = INTIAL_STATE, action) {
    switch (action.type) {
        case SHOW_TOP_ALERT:
            return {
                ...state,
                alertType: action.alertType,
                text: action.text,
                visible: true,
            };
        case HIDE_TOP_ALERT:
            return {
                ...state,
                alertType: "warning",
                text: "",
                visible: false,
            };

        default:
            return state;
    }
}
