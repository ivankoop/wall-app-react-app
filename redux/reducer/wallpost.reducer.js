import {
    GET_WALL_POST_REQUEST,
    GET_WALL_POST_SUCCESS,
    GET_WALL_POST_FAILURE,
    INSERT_WALL_POST,
    DELETE_WALL_POST,
} from "../constants/wallpost.constants";

const INTIAL_STATE = {
    wallPosts: [],
    wallPostsLoading: false,
};

export function wallPostsControls(state = INTIAL_STATE, action) {
    switch (action.type) {
        case GET_WALL_POST_REQUEST:
            return {
                ...state,
                wallPostsLoading: true,
            };
        case GET_WALL_POST_SUCCESS:
            return {
                ...state,
                wallPostsLoading: false,
                wallPosts: action.wallPosts,
            };

        case GET_WALL_POST_FAILURE:
            return {
                ...state,
                wallPostsLoading: false,
            };

        case INSERT_WALL_POST:
            return {
                ...state,
                wallPosts: [action.post].concat(state.wallPosts),
            };

        case DELETE_WALL_POST:
            return {
                ...state,
                wallPosts: state.wallPosts.filter(
                    ({_id}) => _id !== action.postId
                ),
            };

        default:
            return state;
    }
}
