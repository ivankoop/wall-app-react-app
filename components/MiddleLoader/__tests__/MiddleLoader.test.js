import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {combineReducers, createStore} from "redux";
import MiddleLoader from "../MiddleLoader";
import {middleLoaderControls} from "../reducer/middleloader.reducer";

const myReducers = combineReducers({
    middleLoaderControls,
});

const store = createStore(myReducers);

describe("MiddleLoader", () => {
    beforeEach(() => {
        console.error = jest.fn();
    });

    it("renders MiddleLoader", () => {
        jest.useFakeTimers();
        const tree = renderer
            .create(
                <Provider store={store}>
                    <MiddleLoader visible={true} />
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
