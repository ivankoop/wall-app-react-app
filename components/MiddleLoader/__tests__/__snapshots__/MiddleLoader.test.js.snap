// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MiddleLoader renders MiddleLoader 1`] = `null`;
